search/search_quality/clusterize-tag-values.lisp

```diff
@@ -0,0 +1,133 @@
+#|
+exec /usr/bin/env sbcl --noinform --quit --eval "(defparameter *script-name* \"$0\")" --load "$0" --end-toplevel-options "$@"
+|#
+
+;;; This script clusterizes values from the taginfo database and
+;;; prints information about clusters.
+
+;;; Silently loads sqlite.
+(with-open-file (*standard-output* "/dev/null"
+                                   :direction :output
+                                   :if-exists :supersede)
+  (ql:quickload "sqlite"))
+
+(defun latin-char-p (char)
+  (or (and (char>= char #\a) (char<= char #\z))))
+
+(defun get-postcode-pattern (postcode)
+  "Simplifies postcode in the following way:
+   * all latin letters are replaced by 'A'
+   * all digits are replaced by 'N'
+   * hyphens and dots are replaced by a space
+   * other characters are capitalized
+
+   This format follows https://en.wikipedia.org/wiki/List_of_postal_codes.
+  "
+  (map 'string #'(lambda (c) (cond ((latin-char-p c) #\A)
+                                   ((digit-char-p c) #\N)
+                                   ((or (char= #\- c) (char= #\. c)) #\Space)
+                                   (T c)))
+       (string-upcase postcode)))
+
+
+(defun get-phone-or-flat-pattern (phone)
```

Comment:
In the function name or in patterns?

I want to know which separators are used in flats.